src/helpers/wallet-precalculation.helper.js

```diff
@@ -0,0 +1,234 @@
+    'monster opinion bracket aspect mask labor obvious hat matrix exact canoe race shift episode plastic debris dash sort motion juice leg mushroom maximum evidence',
+    'tilt lab swear uncle prize favorite river myth assault transfer venue soap lady someone marine reject fork brain swallow notice glad salt sudden pottery',
+  ],
+  pubkeys: [
+    'xpub6CvvCBtHqFfErbcW2Rv28TmZ3MqcFuWQVKGg8xDzLeAwEAHRz9LBTgSFSj7B99scSvZGbq6TxAyyATA9b6cnwsgduNs9NGKQJnEQr3PYtwK',
+    'xpub6CA16g2qPwukWAWBMdJKU3p2fQEEi831W3WAs2nesuCzPhbrG29aJsoRDSEDT4Ac3smqSk51uuv6oujU3MAAL3d1Nm87q9GDwE3HRGQLjdP',
+    'xpub6BwNT613Vzy7ARVHDEpoX23SMBEZQMJXdqTWYjQKvJZJVDBjEemU38exJEhc6qbFVc4MmarN68gUKHkyZ3NEgXXCbWtoXXGouHpwMEcXJLf',
+    'xpub6DCyPHg4AwXsdiMh7QSTHR7afmNVwZKHBBMFUiy5aCYQNaWp68ceQXYXCGQr5fZyLAe5hiJDdXrq6w3AXzvVmjFX9F7EdM87repxJEhsmjL',
+    'xpub6CgPUcCCJ9pAK7Rj52hwkxTutSRv91Fq74Hx1SjN62eg6Mp3S3YCJFPChPaDjpp9jCbCZHibBgdKnfNdq6hE9umyjyZKUCySBNF7wkoG4uK',
+  ],
+  walletConfig: {
+    pubkeys: [],
+    total: 5,
+    minSignatures: 3,
+  }
+};
+multisigWalletsData.walletConfig.pubkeys = multisigWalletsData.pubkeys;
+
+export class WalletPrecalculationHelper {
+  WALLETS_FILENAME = '';
+
+  walletsDb = [];
+
+  /**
+   * Initializes the helper with a filename to sync the local wallet storage with.
+   * @param walletsFilename
+   */
+  constructor(walletsFilename) {
+    this.WALLETS_FILENAME = walletsFilename || './tmp/wallets.txt';
+  }
+
+  /**
+   * Generates 22 addresses for a wallet 24-word seed.
+   * @param [params]
+   * @param {string} [params.words] Optional wallet seed words. If empty, generates a new wallet
+   * @param {number} [params.addressIntervalStart=0] Optional interval start index
+   * @param {number} [params.addressIntervalEnd=22] Optional interval end index
+   * @param {{minSignatures:number,wordsArray:string[]}} [params.multisig] Optional multisig object
+   * @returns {{addresses: string[], words: string}}
+   */
+  static generateWallet(params = {}) {
+    const timeStart = Date.now().valueOf();
+    const walletCfg = {
+      seed: params.words || '',
+      connection: {},
+      password: '123',
+      pinCode: '123',
+      multisig: false
+    };
+
+    walletCfg.connection = new Connection({
+      network: config.network,
+      servers: [config.server],
+      connectionTimeout: config.connectionTimeout,
+    });
+
+    let multisigDebugData = null;
+    if (params.multisig) {
+      const pubkeys = params.multisig.wordsArray.map(w => walletUtils.getMultiSigXPubFromWords(w));
+      walletCfg.multisig = {
+        pubkeys,
+        minSignatures: params.multisig.minSignatures
+      };
+      multisigDebugData = {
+        words: params.multisig.wordsArray,
+        total: pubkeys.length,
+        ...walletCfg.multisig
+      };
+    } else if (!params.words) {
+      walletCfg.seed = wallet.generateWalletWords();
+    }
+
+    const store = new MemoryStore();
+    Storage.setStore(store);
+    wallet.executeGenerateWallet(
```

Comment:
This won't work for a multisig wallet since the wallet expects the `wallet:multisig` to be set to true

Reply:
Fixed on [264fc88](https://github.com/HathorNetwork/hathor-wallet-headless/pull/189/commits/264fc8801a5027dcd1ee0a5c4cef27a364da2f0f), along with the pre-calculated addresses on `precalculated-multisig-wallets.json`.